### PR TITLE
Fix/pr 544 my mooc feature add onInit callback

### LINKED ISSUE #514: Blank carousel while loading
Hi,

When the carousel is loading, we can see that the space it should be filling is going blank.
In this respect, when making our own SSR implementation of the slider we can see that there is a moment when the carousel is blank.

This following video should be more explicit:

[Carousel](https://vimeo.com/426259397 "Carousel")

Do you have any ideas on how to fix this? I didn't experience this issue while using other lib (such as `react-slick`) but we decided to switch to `react-carousel` ;)

Thank you :rocket: !

### PR DESCRIPTION
**Pull request by piotr-s-brainhub**
Deployed to https://beghp.github.io/gh-pages-rc-2<br>It fixes #514 <br><br>- [x] an issue linked to the PR<br><br>

### CODE REVIEW

**Comment by piotr-s-brainhub** [comment]
I checked the callback is called and this PR doesn't break anything (desktop).
Screenshot 2020-06-18 at 16 34 34

File: src/components/Carousel.js [hunk 1]
```diff
@@ -105,6 +106,10 @@ class Carousel extends Component {
     this.onResize();
     // setting autoplay interval
     this.resetInterval();
+
+    if (typeof this.props.handleIsReady !== 'undefined') {
+      this.props.handleIsReady();
+    }
```
Comment (RobertHebel) [inline src/components/Carousel.js:112]:
```suggestion
    if (typeof this.props.handleIsReady === 'function') {
      this.props.handleIsReady();
    }
```

If it has to be function perhaps it's a better idea to check type explicitly. WDY?

File: README.md [hunk 2]
```diff
@@ -172,6 +172,7 @@ You can access a clickable demo with many examples and a [live code editor](http
 | [**clickToChange**](https://brainhubeu.github.io/react-carousel/docs/examples/clickToChange) | *Boolean* | `undefined` | Clicking on a slide changes current slide to the clicked one |
 | [**dots**](https://brainhubeu.github.io/react-carousel/docs/examples/simpleDots) | *Boolean* | `undefined` | Renders default dots under the carousel |
 | [**draggable**](https://brainhubeu.github.io/react-carousel/docs/examples/draggable) | *Boolean* | `true` | Makes it possible to drag to the next slide with mouse cursor |
+| **handleIsReady** | *Function* | `undefined` | Callback thrown after the carousel is loaded |
```
Comment (RobertHebel) [inline README.md:175]:
```suggestion
| **onInit** | *Function* | `undefined` | Callback thrown after the carousel is loaded |
```

I think it's good idea to have all callback functions start with `on` prefix

File: docs/api/carousel.md [hunk 3]
```diff
@@ -40,6 +40,8 @@
 
 * ```minDraggableOffset: Number``` Defines the minimum offset to consider the drag gesture
 
+* ```handleIsReady: Function``` Callback thrown after the carousel is loaded
```
Comment (RobertHebel) [inline docs/api/carousel.md:43]:
```suggestion
* ```onInit: Function``` Callback thrown after the carousel is loaded
```

File: src/components/Carousel.js [hunk 4]
```diff
@@ -42,6 +42,7 @@ class Carousel extends Component {
     dots: PropTypes.bool,
     className: PropTypes.string,
     minDraggableOffset: PropTypes.number,
+    handleIsReady: PropTypes.func,
```
Comment (RobertHebel) [inline src/components/Carousel.js:45]:
```suggestion
    onInit: PropTypes.func,
```

**Comment by brainhub-devops-2** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/uglify-js/node_modules/wordwrap
 | LICENSE : MIT*
 | LICENSE_FILE : node_modules/uglify-js/node_modules/wordwrap/README.markdown
 | REPOSITORY: https://github.com/substack/node-wordwrap
 | PUBLISHER : James Halliday
 | EMAIL : mail@substack.net
 | URL : http://substack.net

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/vargs
 | LICENSE : MIT*
 | LICENSE_FILE : node_modules/vargs/LICENSE
 | REPOSITORY: undefined
 | PUBLISHER : Alexis Sellier
 | EMAIL : self@cloudhead.net
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/gemini-coverage/node_modules/uglify-js
 | LICENSE : BSD*
 | LICENSE_FILE : node_modules/gemini-coverage/node_modules/uglify-js/LICENSE
 | REPOSITORY: https://github.com/mishoo/UglifyJS2
 | PUBLISHER : undefined
 | EMAIL : undefined
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/trim
 | LICENSE : MIT*
 | LICENSE_FILE : node_modules/trim/Readme.md
 | REPOSITORY: undefined
 | PUBLISHER : TJ Holowaychuk
 | EMAIL : tj@vision-media.ca
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/istanbul/node_modules/source-map
 | LICENSE : BSD
 | LICENSE_FILE : node_modules/istanbul/node_modules/source-map/LICENSE
 | REPOSITORY: https://github.com/mozilla/source-map
 | PUBLISHER : Nick Fitzgerald
 | EMAIL : nfitzgerald@mozilla.com
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/gemini-coverage/node_modules/source-map
 | LICENSE : BSD
 | LICENSE_FILE : node_modules/gemini-coverage/node_modules/source-map/LICENSE
 | REPOSITORY: https://github.com/mozilla/source-map
 | PUBLISHER : Nick Fitzgerald
 | EMAIL : nfitzgerald@mozilla.com
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/optimist
 | LICENSE : MIT*
 | LICENSE_FILE : node_modules/optimist/LICENSE
 | REPOSITORY: https://github.com/substack/node-optimist
 | PUBLISHER : James Halliday
 | EMAIL : mail@substack.net
 | URL : http://substack.net

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/map-stream
 | LICENSE : Custom: https://github.com/dominictarr/event-stream
 | LICENSE_FILE : node_modules/map-stream/LICENCE
 | REPOSITORY: https://github.com/dominictarr/map-stream
 | PUBLISHER : Dominic Tarr
 | EMAIL : dominic.tarr@gmail.com
 | URL : http://dominictarr.com

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/jsonify
 | LICENSE : Public Domain
 | LICENSE_FILE : node_modules/jsonify/README.markdown
 | REPOSITORY: https://github.com/substack/jsonify
 | PUBLISHER : Douglas Crockford
 | EMAIL : undefined
 | URL : http://crockford.com/

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/json-schema
 | LICENSE : AFLv2.1,BSD
 | LICENSE_FILE : node_modules/json-schema/README.md
 | REPOSITORY: https://github.com/kriszyp/json-schema
 | PUBLISHER : Kris Zyp
 | EMAIL : undefined
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/indexof
 | LICENSE : MIT*
 | LICENSE_FILE : node_modules/indexof/Readme.md
 | REPOSITORY: undefined
 | PUBLISHER : undefined
 | EMAIL : undefined
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/html
 | LICENSE : BSD*
 | LICENSE_FILE : node_modules/html/LICENSE
 | REPOSITORY: https://github.com/maxogden/commonjs-html-prettyprinter
 | PUBLISHER : Max Ogden
 | EMAIL : max@maxogden.com
 | URL : http://maxogden.com

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/glob-to-regexp
 | LICENSE : BSD*
 | LICENSE_FILE : node_modules/glob-to-regexp/README.md
 | REPOSITORY: https://github.com/fitzgen/glob-to-regexp
 | PUBLISHER : Nick Fitzgerald
 | EMAIL : fitzgen@gmail.com
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/uglifyify/node_modules/extend
 | LICENSE : MIT*
 | LICENSE_FILE : node_modules/uglifyify/node_modules/extend/LICENSE
 | REPOSITORY: https://github.com/justmoon/node-extend
 | PUBLISHER : Stefan Thomas
 | EMAIL : justmoon@members.fsf.org
 | URL : http://www.justmoon.net

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/istanbul/node_modules/estraverse
 | LICENSE : BSD
 | LICENSE_FILE : node_modules/istanbul/node_modules/estraverse/LICENSE.BSD
 | REPOSITORY: https://github.com/estools/estraverse
 | PUBLISHER : undefined
 | EMAIL : undefined
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/domutils
 | LICENSE : BSD*
 | LICENSE_FILE : node_modules/domutils/LICENSE
 | REPOSITORY: https://github.com/FB55/domutils
 | PUBLISHER : Felix Boehm
 | EMAIL : me@feedic.com
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/eslint-plugin-import/node_modules/doctrine
 | LICENSE : BSD
 | LICENSE_FILE : node_modules/eslint-plugin-import/node_modules/doctrine/LICENSE.BSD
 | REPOSITORY: https://github.com/eslint/doctrine
 | PUBLISHER : undefined
 | EMAIL : undefined
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/css-select
 | LICENSE : BSD*
 | LICENSE_FILE : node_modules/css-select/LICENSE
 | REPOSITORY: https://github.com/fb55/css-select
 | PUBLISHER : Felix Boehm
 | EMAIL : me@feedic.com
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/parse-color/node_modules/color-convert
 | LICENSE : MIT*
 | LICENSE_FILE : node_modules/parse-color/node_modules/color-convert/LICENSE
 | REPOSITORY: https://github.com/harthur/color-convert
 | PUBLISHER : Heather Arthur
 | EMAIL : fayearthur@gmail.com
 | URL : undefined

  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  MODULE : node_modules/async-foreach
 | LICENSE : MIT*
 | LICENSE_FILE : node_modules/async-foreach/LICENSE-MIT
 | REPOSITORY: https://github.com/cowboy/javascript-sync-async-foreach
 | PUBLISHER : "Cowboy" Ben Alman
 | EMAIL : undefined
 | URL : http://benalman.com/

  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 1ae609b8616a6a2ad4b437159450b72c87cb31b9
</p>

**Comment by piotr-s-brainhub** [comment]
I checked it's still working
```
<Carousel onInit={(...args)=>{console.log('onInit', args)}}>
  <img src={imageOne} />
  <img src={imageTwo} />
  <img src={imageThree} />
</Carousel>
```
Screenshot 2020-07-10 at 17 27 43

and it doesn't break anything (desktop).

**Pull request merged**